### PR TITLE
KAZOO-4470: Run start_dtmf in case of "101 telephone-event" absence.

### PR DESCRIPTION
**Pull request by onnet**
Reopen of https://github.com/2600hz/kazoo/pull/1510

### CODE REVIEW

File: applications/ecallmgr/src/ecallmgr_fs_route.erl [hunk 1]
```diff
@@ -278,7 +278,9 @@ search_for_route(Section, Node, FetchId, CallId, Props) ->
             lager:info("did not receive route response for request ~s: ~p", [FetchId, _R]);
         {'ok', JObj} ->
             'true' = wapi_route:resp_v(JObj),
-            J = wh_json:set_value(<<"Context">>, hunt_context(Props), JObj),
+            Routines = [fun(S) -> wh_json:set_value(<<"Context">>, hunt_context(Props), S) end
+                        ,fun(S) -> wh_json:set_value(<<"Remote-SDP">>, hunt_rsdp(Props), S) end],
```
Comment (lazedo) [inline applications/ecallmgr/src/ecallmgr_fs_route.erl]: @onnet can we change `hunt_rsdp` to `remote_sdp` ?

File: applications/ecallmgr/src/ecallmgr_fs_route.erl [hunk 2]
```diff
@@ -327,14 +333,26 @@ reply_forbidden(Section, Node, FetchId) ->
...
             ecallmgr_fs_channels:update(CallId, #channel.handling_locally, 'true'),
-            maybe_start_call_handling(Node, FetchId, CallId, JObj)
+            maybe_start_call_handling(Node, FetchId, CallId, J)
+    end.
+
+-spec check_dtmf_type(ne_binary()) -> ne_binary().
+check_dtmf_type(JObj) ->
+    SDPRemote = wh_json:get_value(<<"Remote-SDP">>, JObj, <<"101 telephone-event">>),
+    case binary:match(SDPRemote, <<"101 telephone-event">>) of
+        'nomatch' -> wh_json:set_value(<<"Pre-Park">>, <<"start_dtmf">>, JObj);
```
Comment (lazedo) [inline applications/ecallmgr/src/ecallmgr_fs_route.erl]: @onnet this may override already set  `<<"Pre-Park">>`

Comment (onnet) [inline applications/ecallmgr/src/ecallmgr_fs_route.erl]: Yep, this is actually the goal of this PR...

Comment (lazedo) [inline applications/ecallmgr/src/ecallmgr_fs_route.erl]: @onnet but why are you doing this ? can't `ring_ready` and `start_dtmf` coexist ?

Comment (onnet) [inline applications/ecallmgr/src/ecallmgr_fs_route.erl]: @lazedo hm.. looks like they could. Didn't pay attention to this fact before :)
This means that we can call it one after another.

Comment (onnet) [inline applications/ecallmgr/src/ecallmgr_fs_route.erl]: @lazedo next iteration is ready :)

**Comment by jamesaimonetti** [comment]
@onnet thanks! I've made a few style changes; let me know if you have any questions about them.

**Pull request merged**